src/EditorFeatures/Next/Options/EditorConfigDocumentOptionsProvider.cs

```diff
@@ -80,8 +77,10 @@ private void Workspace_DocumentOpened(object sender, DocumentEventArgs e)
                     TaskContinuationOptions.OnlyOnRanToCompletion | TaskContinuationOptions.ExecuteSynchronously,
                     TaskScheduler.Default);
 
-                var context = await cancellableContextTask.ConfigureAwait(false);
-                return new DocumentOptions(context.CurrentConventions);
+                using (var context = await cancellableContextTask.ConfigureAwait(false))
```

Comment:
Why are we disposing the context now? We dispose this in Workspace_DocumentClosed. Isn't this creating a race?